src/entities/user/service/user.service.ts

```diff
@@ -0,0 +1,30 @@
+import {Injectable} from '@nestjs/common';
+import {UserDocument} from '../model/user.schema';
+import {InjectModel} from '@nestjs/mongoose';
+import {Model} from 'mongoose';
+import * as bcrypt from 'bcrypt';
+import {CreateUserDto} from '../dto/createUserDto';
+
+@Injectable()
+export class UserService {
+  constructor(@InjectModel('User') private userModel: Model<UserDocument>) {}
+
+  async findOne(email: string): Promise<UserDocument> {
+    return this.userModel.findOne({email: email});
+  }
+
+  async createUser(userData: CreateUserDto): Promise<UserDocument> {
+    const {firstName, lastName, email, phoneNumber, password} = userData;
+    const saltOrRounds = 10;
+    const hash = await bcrypt.hash(password, saltOrRounds);
+    const newUser = new this.userModel({
+      firstName,
+      lastName,
+      email,
+      phoneNumber,
+      passwordHash: hash
```

Comment:
It potentially can be implemented via setter. When you will hash it in a setter and here just pass a password.

Reply:
Ok, thanks. Where to place this file with setter better? I assume I don`t want to have it directly in user.schema.ts. So I can create separate file like setPassword.ts in a model folder?

Reply:
A problem here. The password is hashed asynchronously. Setters works synchronous only. I will try to solve.

Reply:
Solved.
```
UserSchema.pre('save', async function (next) {
  if (this.isModified('password')) {
    this.password = await setPassword(this.password);
  }
  next();
});
```

Reply:
Great.